component/module/git/git.go

```diff
@@ -0,0 +1,269 @@
+// Package git implements the module.git component.
+package git
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/module"
+	"github.com/grafana/agent/component/module/git/internal/vcs"
+	"github.com/grafana/agent/pkg/river"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "module.git",
+		Args:    Arguments{},
+		Exports: module.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the module.git component.
+type Arguments struct {
+	Repository string `river:"repository,attr"`
+	Revision   string `river:"revision,attr,optional"`
+	Path       string `river:"path,attr"`
+
+	PullFrequency time.Duration `river:"pull_frequency,attr,optional"`
+
+	Arguments map[string]any `river:"arguments,block,optional"`
+}
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	Revision:      "HEAD",
+	PullFrequency: time.Minute,
+}
+
+var _ river.Unmarshaler = (*Arguments)(nil)
+
+// UnmarshalRiver implements river.Unmarshaler and applies default settings to
+// args.
+func (args *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*args = DefaultArguments
+
+	type arguments Arguments
+	return f((*arguments)(args))
+}
+
+// Component implements the module.git component.
+type Component struct {
+	opts component.Options
+	log  log.Logger
+	mod  *module.ModuleComponent
+
+	mut      sync.RWMutex
+	repo     *vcs.GitRepo
+	repoOpts vcs.GitRepoOptions
+	args     Arguments
+
+	argsChanged chan struct{}
+
+	healthMut sync.RWMutex
+	health    component.Health
```

Comment:
Do we need to track a second health here or can we use the health on module.ModuleComponent? If that implementation is unclear maybe we update it but the intention is for module.ModuleComponent.health to manage the module health.

related, updateHealth looks like it could be common enough we should add it to module.go instead of in here.

Reply:
We need a second health for the health of the git cloning code, yes. Actually, let me update this to use the new `component.LeastHealthy` functionality

Reply:
sounds good, this will make more sense to track separate in that context I think.